gnatsd/check.py

```diff
@@ -0,0 +1,150 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import requests
+
+# stdlib
+
+# project
+from checks import AgentCheck
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'gnatsd'
+
+class GnatsdCheck(AgentCheck):
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.counts = {}
+
+    def check(self, instance):
+        self.InstanceCheck(instance, self).check()
+
+    class InstanceCheck:
```

Comment:
Can you merge this into the parent class? This class doesn't seem to add anything to the parent class other than another level of indirection.

Reply:
The `GnatsdCheck` is a "singleton" that receives multiple calls to the `check` method with an _instance_ object.   Then we do multiple calls that check different things derived from that `instance` object.  This leaves us passing the `instance` argument around to every method and keeping track of the state in various method arguments.  It's really very complicated and procedural to maintain the state like that.   We found it much more straightforward to keep the "per call" state in an instance of an object (the `InstanceCheck`) and encapsulate the state of individual `check` calls outside of the GnatdsCheck singleton.

Is there a reason you'd prefer the more procedural style?

Reply:
Well, it's the way we do all of the other checks so it's a bit easier to follow. However, we've allowed people to use this style before. Instead of doing it like this, though, can you change a few things for me? First, can you give the class a better name. Second can you make it a top level class? The nested class is much harder to follow.

And it's the same with the config, just give it a better name and put it at the top level in the file. While I think using a dict for config would be a bit simpler, I don't really oppose this style of writing a check.